modules/library/main/src/main/java/org/geotools/styling/visitor/RenderingSelectorStyleVisitor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.styling.visitor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.geotools.styling.FeatureTypeStyle;
+import org.geotools.styling.Rule;
+import org.geotools.styling.Symbolizer;
+
+/**
+ * This class applies the renderingLegend and renderingMap VendorOptions to a Style eg.
+ * <VendorOption name="renderingLegend">false</VendorOption> <VendorOption
+ * name="renderingMap">false</VendorOption> The visitor avoid including in the deep copy of the
+ * Style being passed to it the FeatureTypeStyle or the Rule or the Symbolizer for which the
+ * vendorOptions passed to the constructor has been defined with a false value.
+ */
+public class RenderingSelectorStyleVisitor extends DuplicatingStyleVisitor {
+
+    public static final String RENDERING_MAP_OPTION = "renderingMap";
```

Comment:
We normally place vendor option names in the interfaces of the receiving object, e.g.:

* [TextSymbolizer](https://github.com/geotools/geotools/blob/f66689b5305eaf381e21dc2feec8011138d4cb24/modules/library/main/src/main/java/org/geotools/styling/TextSymbolizer.java#L81)
* [FeatureTypeStyle](https://github.com/geotools/geotools/blob/f66689b5305eaf381e21dc2feec8011138d4cb24/modules/library/main/src/main/java/org/geotools/styling/FeatureTypeStyle.java#L64)
* [PolygonSymbolizer](https://github.com/geotools/geotools/blob/f66689b5305eaf381e21dc2feec8011138d4cb24/modules/library/main/src/main/java/org/geotools/styling/PolygonSymbolizer.java#L73)

In particular, in the one contained in the "orgo.geootols" packages, as they are vendor options.

Reply:
fixed, I put it on the FeatureTypeStyle interface